web/src/components/DataTable.js

```diff
@@ -89,101 +99,240 @@ function ContractList({
+      rowData[objIndex][key] = value;
+    }
+    setIsUpdate(true);
+  };
+
+  const handleChangeTextfield = (key, value, options) => {
+    const objIndex = rowData.indexOf(options);
+    if (objIndex !== -1) {
+      rowData[objIndex][key] = value;
     }
+    setIsUpdate(true);
   };
+  const handleClickLink = (rowData) => console.log('rowData...', rowData);
 
   // HandleChange body
   const handleChangeBody = (options, idx) => {
     const key = Object.keys(options)[idx];
     const newVal = { key, value: options[key] };
     switch (headCellsType[idx]) {
       case 'BUTTON':
-        return <Button variant="contained">{newVal.value}</Button>;
+        return (
+          <Tooltip title={newVal.value}>
+            <Button variant="contained" tooltip={newVal.value}>
+              {newVal.value}
+            </Button>
+          </Tooltip>
+        );
+      case 'NONE':
+        return (
+          <Tooltip title={newVal?.value}>
+            <Typography style={{ fontSize: '13px' }}>{newVal?.value}</Typography>
+          </Tooltip>
+        );
+      case 'TEXTFIELD':
+        if (newVal) {
+          return (
+            <Grid style={{ marginTop: '0.1rem', marginBottom: '0.1rem' }}>
+              <RenderComponent
+                metaData={{
+                  control: TEXT_FIELD,
+                  key: `${newVal?.key}`,
+                  label: `${newVal?.key}`,
+                  placeholder: `${newVal?.key}`,
+                  columnWidth: 12,
+                  size: 'large'
+                }}
+                payload={{ [newVal?.key]: newVal?.value }}
+                ind={1}
+                handleChange={(key, val) => handleChangeTextfield(key, val, options)}
+              />
+            </Grid>
+          );
+        }
+        break;
+      case 'LINK':
+        return (
+          <Tooltip title={newVal.value}>
+            <Typography
+              style={{ cursor: 'pointer', textDecoration: 'underline', fontSize: '13px', color: 'blue' }}
+              onClick={() => handleClickLink(options)}
+            >
+              {newVal.value}
+            </Typography>
+          </Tooltip>
+        );
+      case 'DATE':
+        return (
+          <div style={{ marginTop: '0.5rem', marginBottom: '0.5rem' }}>
+            <RenderComponent
+              metaData={{
+                control: DATEPICKER,
+                key: `${newVal?.key}`,
+                label: `${newVal?.key}`,
+                placeholder: `${newVal?.key}`,
+                columnWidth: 12,
+                size: 'large'
+              }}
+              payload={{ [newVal?.key]: newVal?.value }}
+              ind={1}
+              handleChange={(key, val) => handleChangeDate(key, val, options)}
+            />
+          </div>
+        );
+      default:
+        return undefined;
+    }
+  };
+  // Handle ExpandedDatechange
+  const handleExpandedChangeDate = (key, value, options) => {
+    // console.log('Logging...', newValue, data, key);
```

Comment:
removed unnecessary commented code

Reply:
updated

Reply:
not resolved

Reply:
that colsole line is removed I'll commit again